lib/Middleware/CanSeeLoginPageMiddleware.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Benjamin Gaussorgues <benjamin.gaussorgues@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\LimitLoginToIp\Middleware;
+
+use Exception;
+use OC\Core\Controller\LoginController;
+use OCA\LimitLoginToIp\Exception\ForbiddenLoginException;
+use OCA\LimitLoginToIp\IsRequestAllowed;
+use OCP\AppFramework\Middleware;
+use OCP\IURLGenerator;
+
+/**
+ * Checks whether the "sharing check" is enabled
+ *
+ * @package OCA\LimitLoginToIp\Middleware
+ * @psalm-api
+ */
+class CanSeeLoginPageMiddleware extends Middleware {
+	private const LOGIN_METHOD_NAMES = [
+		'tryLogin',
+		'confirmPassword',
+		'showLoginForm',
```

Comment:
But yourself as a code owner on that file to be aware of changes? :D

Reply:
Good idea!